api/src/context.rs

```diff
@@ -105,17 +105,16 @@ impl Context {
         self.node_config.api.max_account_modules_page_size
     }
 
-    pub fn move_resolver(&self) -> Result<StorageAdapterOwned<DbStateView>> {
-        self.db
-            .latest_state_checkpoint_view()
-            .map(|state_view| state_view.into_move_resolver())
```

Comment:
why remove this helper function? we can't return something like state_view.as_move_resolver?

Reply:
My point is that we should never need a MoveResolver by value. All functions should take a MoveResolver by reference and we can on the fly convert (StorageAdaptor is just a wrapper around a StateView reference)

Reply:
we can still have a helper function that returns &impl MoveResolver though?

Reply:
Yes indeed